libstuff/STCPNode.cpp

```diff
@@ -234,20 +276,95 @@ void STCPNode::postPoll(fd_map& fdm, uint64_t& nextActivity) {
                 peer->reset();
                 peer->s = openSocket(peer->host);
                 if (peer->s) {
-                    // Try to log in now.  Send a PING immediately after so we
-                    // can get a fast estimate of latency.
-                    SData login("NODE_LOGIN");
-                    login["Name"] = name;
-                    peer->s->send(login.serialize());
-                    _sendPING(peer);
-                    _onConnect(peer);
+                    if(!peer->s->ssl) {
+                            
+                        /*********** TLS NODE CLIENT ***********/
+
+                        // Escalate to SSL?
+                        SX509* x509 = new SX509;
+                        SDEBUG("MB Opening TLS Client");
+                        x509 = SX509Open("", "", "", false, "");
+                        peer->ssl = SSSLOpen(peer->s->s,x509,false);
+                        string domain;
+                        uint16_t serverport = 0;
+                        if (!SParseHost(peer->host, domain, serverport)) {
+                            STHROW("invalid host: " + peer->host);
+                        }
+                        // TODO put our local server name here.
+                        // mbedtls_ssl_set_hostname( &peer->ssl->ssl, "mbed TLS Client 1" );
+
+                        // TODO re-use our existing socket? This feels like duplicated work.
+
+                        mbedtls_net_init( &peer->ssl->ctx );
+
+                        mbedtls_net_connect( &peer->ssl->ctx, domain.c_str(), (char *)&serverport, MBEDTLS_NET_PROTO_TCP );
+                        
+                        int ret = 0;
+                        int tries = 0;
+                        do {
+                            tries++;
+                            ret = mbedtls_ssl_handshake(&peer->ssl->ssl);
+                            SDEBUG("MB Client SSL Handshake " << ret << " : " << SSSLError(ret));
+                            // remove me after debugging.
+                            sleep(1);
+                            if(tries > 100) {
```

Comment:
Why does this happen? Seems like we could return immediately on failure and try again later.